core/src/main/scala/org/apache/spark/storage/BlockId.scala

```diff
@@ -172,11 +195,15 @@ object BlockId {
   val SHUFFLE_BATCH = "shuffle_([0-9]+)_([0-9]+)_([0-9]+)_([0-9]+)".r
   val SHUFFLE_DATA = "shuffle_([0-9]+)_([0-9]+)_([0-9]+).data".r
   val SHUFFLE_INDEX = "shuffle_([0-9]+)_([0-9]+)_([0-9]+).index".r
-  val SHUFFLE_PUSH = "shufflePush_([0-9]+)_([0-9]+)_([0-9]+)".r
-  val SHUFFLE_MERGED_DATA = "shuffleMerged_([_A-Za-z0-9]*)_([0-9]+)_([0-9]+).data".r
-  val SHUFFLE_MERGED_INDEX = "shuffleMerged_([_A-Za-z0-9]*)_([0-9]+)_([0-9]+).index".r
-  val SHUFFLE_MERGED_META = "shuffleMerged_([_A-Za-z0-9]*)_([0-9]+)_([0-9]+).meta".r
-  val SHUFFLE_CHUNK = "shuffleChunk_([0-9]+)_([0-9]+)_([0-9]+)".r
+  val SHUFFLE_PUSH = "shufflePush_([0-9]+)_(-?[0-9]+)_([0-9]+)_([0-9]+)".r
```

Comment:
@Ngone51 This is something I want to discuss and get some thoughts on. For deterministic stage, we will keep `shuffleSequenceId` to -1 this way we can distinguish between deterministic stage and indeterministic stage, so that may be if needed we can clean up the entry in the Map in `RemoteBlockPushResolver` as well as for cases in the future where we might need to distinguish between the 2. Any thoughts?

Reply:
I see. Sounds fine.

Reply:
Currently, in this PR, we are not handling this as stated above.
Even if we were to do it, I would prefer to use `0` for deterministic stages - and `> 0` for others : including the very first stage attempt execution for a non-deterministic stage.

Reply:
Yes we are not cleaning up the metadata from `RemoteBlockPushResolver` for deterministic totally from the Map but will do it as a follow-up.

Reply:
If we are not planning to do this right now, remove need for negative values and treat as 0 vs non-zero positive ?

Reply:
Sure. Also I made the changes for cleanup in the cases of determinate shuffles.